pennylane_lightning/core/src/simulators/lightning_tensor/tncuda/gates/tests/Test_TNCuda_NonParam.cpp

```diff
@@ -0,0 +1,694 @@
+    using cp_t = typename TNDevice_T::ComplexT;
+    std::size_t num_qubits = 3;
+    std::size_t maxExtent = 2;
+
+    std::unique_ptr<TNDevice_T> tn_state =
+        createTNState<TNDevice_T>(num_qubits, maxExtent);
+
+    SECTION("Apply different wire indices") {
+        const std::size_t index = GENERATE(0, 1, 2);
+
+        tn_state->applyOperation("Hadamard", {index}, inverse);
+
+        tn_state->applyOperation("Identity", {index}, inverse);
+        cp_t expected(1.0 / std::sqrt(2), 0);
+
+        tn_state_append_mps_final_state(tn_state);
+
+        auto results = tn_state->getDataVector();
+
+        CHECK(expected == Pennylane::Util::approx(
+                              results[0b1 << ((num_qubits - 1 - index))]));
+    }
+}
+
+TEMPLATE_LIST_TEST_CASE("TNCuda::Gates::Hadamard", "[TNCuda_Nonparam]",
+                        TestTNBackends) {
+    const bool inverse = GENERATE(false, true);
+    using TNDevice_T = TestType;
+    using cp_t = typename TNDevice_T::ComplexT;
+    std::size_t num_qubits = 3;
+    std::size_t maxExtent = 2;
+
+    std::unique_ptr<TNDevice_T> tn_state =
+        createTNState<TNDevice_T>(num_qubits, maxExtent);
+
+    SECTION("Apply different wire indices") {
+        const std::size_t index = GENERATE(0, 1, 2);
+
+        tn_state_append_mps_final_state(tn_state);
+
+        tn_state->applyOperation("Hadamard", {index}, inverse);
+
+        tn_state_append_mps_final_state(tn_state);
+
+        tn_state->applyOperation("Identity", {index}, inverse);
+
+        // Test for multiple final states appendings
+        tn_state_append_mps_final_state(tn_state);
+
+        cp_t expected(1.0 / std::sqrt(2), 0);
+
+        auto results = tn_state->getDataVector();
+
+        CHECK(expected == Pennylane::Util::approx(
+                              results[0b1 << ((num_qubits - 1 - index))]));
+    }
+}
+
+TEMPLATE_LIST_TEST_CASE("TNCuda::Gates::PauliX", "[TNCuda_Nonparam]",
+                        TestTNBackends) {
+    const bool inverse = GENERATE(false, true);
+    using TNDevice_T = TestType;
+    using cp_t = typename TNDevice_T::ComplexT;
+
+    std::size_t num_qubits = 3;
+    std::size_t maxExtent = 2;
+
+    std::unique_ptr<TNDevice_T> tn_state =
+        createTNState<TNDevice_T>(num_qubits, maxExtent);
+
+    SECTION("Apply different wire indices") {
+        const std::size_t index = GENERATE(0, 1, 2);
+
+        tn_state->applyOperation("PauliX", {index}, inverse);
+
+        tn_state_append_mps_final_state(tn_state);
+
+        auto results = tn_state->getDataVector();
+
+        CHECK(results[0] == cuUtil::ZERO<cp_t>());
+        CHECK(results[0b1 << (num_qubits - index - 1)] == cuUtil::ONE<cp_t>());
+    }
+}
+
+TEMPLATE_LIST_TEST_CASE("TNCuda::Gates::applyOperation-gatematrix",
+                        "[TNCuda_Nonparam]", TestTNBackends) {
+
+    using TNDevice_T = TestType;
+    using cp_t = typename TNDevice_T::ComplexT;
+    // using Precision_T = typename TNDevice_T::PrecisionT;
```

Comment:
Please remove this commented line. Thanks

Reply:
Done!